src/main/java/io/druid/data/input/impl/CustomParseSpec.java

```diff
@@ -0,0 +1,81 @@
+package io.druid.data.input.impl;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.common.parsers.Parser;
+
+import java.lang.reflect.Constructor;
+import java.util.Map;
+
+/**
+ */
+public class CustomParseSpec extends ParseSpec
+{
+
+  private final String parserClass;
+  private final Map<String, Object> properties;
+
+  private final Constructor constructor;
+
+  @JsonCreator
+  public CustomParseSpec(
+      @JsonProperty("timestampSpec") TimestampSpec timestampSpec,
+      @JsonProperty("dimensionsSpec") DimensionsSpec dimensionsSpec,
+      @JsonProperty("parserClass") String parserClass,
+      @JsonProperty("properties") Map<String, Object> properties
+  )
+  {
+    super(timestampSpec, dimensionsSpec);
+
+    this.parserClass = parserClass;
+    this.properties = properties == null ? ImmutableMap.<String, Object>of() : properties;
+    ClassLoader loader = Thread.currentThread().getContextClassLoader();
```

Comment:
this is probably okay for now, but would be nice to get more feedback a way of extending parsers without reflection